amd_openvx_extensions/amd_rpp/source/tensor/TensorAddTensor.cpp

```diff
@@ -0,0 +1,225 @@
+/*
+Copyright (c) 2024 Advanced Micro Devices, Inc. All rights reserved.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#include "internal_publishKernels.h"
+#if _WIN32
+#include <intrin.h>
+#else
+#include <immintrin.h>
+#include <smmintrin.h>
+#include <x86intrin.h>
+#endif
+
+struct TensorAddTensorLocalData {
+    Rpp32u deviceType;
+    RppPtr_t pSrc1;
+    RppPtr_t pSrc2;
+    RppPtr_t pDst;
+    RpptROI *pSrcRoi;
+    RpptROI *pDstRoi;
+    size_t inputTensorDims1[RPP_MAX_TENSOR_DIMS];
+    size_t inputTensorDims2[RPP_MAX_TENSOR_DIMS];
+    vx_enum inTensorType;
+    vx_enum outTensorType;
+};
+
```

Comment:
I think tensorAdd is misleading name for this operator. I am assuming it is doing a scalar(const) add to all elements of an  audio sample.  May be, we need to make it as generic like arithmetic (add, subtract, mul etc) and specify how operation works within the parameters.